nlp_utilities.jl

```diff
@@ -0,0 +1,236 @@
+using JuMP.MOI
+using SparseArrays
+
+"""
+    create_nlp_model(model::JuMP.Model)
+
+Create a Nonlinear Programming (NLP) model from a JuMP model.
+"""
+function create_nlp_model(model::JuMP.Model)
+    rows = Vector{ConstraintRef}(undef, 0)
+    nlp = MOI.Nonlinear.Model()
+    for (F, S) in list_of_constraint_types(model)
+        if F <: VariableRef
+            continue  # Skip variable bounds
+        end
```

Comment:
Should we allow variable-in-EqualTo (e.g. from `fix`) here?

Reply:
That is a good point! I think we should. I will think about how to update things

Reply:
Solved :)